catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java

```diff
@@ -415,17 +419,21 @@ public void run() {
                 }
 
                 if (startDate != null) {
-                    if (startDate.getTime() > lastExecutionTime) {
-                        shouldRun = false;
+                    long now = System.currentTimeMillis();
+                    if (startDate.getTime() > now) {
+                        long waitToStart = startDate.getTime() - now;
+                        long waitSecs = TimeUnit.MILLISECONDS.toSeconds(waitToStart);
+                        LOGGER.debug("Start time for subscription is in the future, waiting {} seconds", waitSecs);
```

Comment:
Good catch. I have moved this debug to the correct location to get an accurate log of how long the wait will be.